src/screens/hotSpots/HotSpots.tsx

```diff
@@ -0,0 +1,82 @@
+import {View, FlatList} from 'react-native';
+import React, {useEffect, useState} from 'react';
+import MealPriceItem from 'components/mealLists/mealPriceItem/MealPriceItem';
+import {Searchbar} from 'react-native-paper';
+import {colors} from 'utils/theme';
+import {Meal} from 'utils/constants';
+import styles from './style';
+import MealService from 'api/MealService';
+import AppBar from 'components/appbar/AppBar';
+import {useNavigation} from '@react-navigation/native';
+import LoadingOverlay from 'components/loadingOverlay/LoadingOverlay';
+
+const HotSpots = () => {
+  const navigation = useNavigation();
+  const [searchQuery, setSearchQuery] = React.useState('');
+  const [meals, setMeals] = useState<Meal[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchMeals = async () => {
+      setLoading(true);
+      try {
+        let data: Meal[] = [];
+        if (searchQuery.trim().length > 0) {
+          // User is searching by name
+          data = await MealService.searchMealsByName(searchQuery);
+        } else {
+          // Default fetch by letter
+          const originalData = await MealService.fetchMealsByLetter('s');
+          data = originalData;
+        }
+        setMeals(data);
+      } catch (error) {
+        console.error('Error fetching meals:', error);
+      } finally {
+        setLoading(false);
+      }
+    };
+    // Triggered whenever searchQuery changes
+    fetchMeals();
+  }, [searchQuery]);
+
+  const handleBackPress = () => {
+    navigation.goBack();
+  };
+
+  return (
+    <View style={styles.container}>
+      <AppBar title="Hot Spots" onBackPress={handleBackPress} />
+      {/* Search View */}
+      <Searchbar
+        placeholder="Search"
+        onChangeText={setSearchQuery}
+        value={searchQuery}
+        style={styles.searchbar}
+      />
+      <View style={styles.hotSpots}>
+        <FlatList
+          data={meals}
+          renderItem={({item}) => (
+            <MealPriceItem
+              image={item.strMealThumb}
+              name={item.strMeal}
+              idMeal={item.idMeal}
+            />
+          )}
+          keyExtractor={item => item.idMeal}
+          horizontal={false}
```

Comment:
no need to write this because flatlist is by default  vertical. 
```suggestion
          horizontal={false}
```